vizro-core/src/vizro/cards/_cards.py

```diff
@@ -0,0 +1,119 @@
+"""Module containing default card components."""
+
+from typing import Callable, Optional
+
+import dash_bootstrap_components as dbc
+import pandas as pd
+from dash import dcc, get_relative_path, html
+
+from vizro.models.types import capture
+
+
+# LQ: The data frame is not used in the function, but needs to be provided here. I guess that is unavoidable for now?
+@capture("card")
+def text_card(data_frame: pd.DataFrame, text: str) -> dbc.Card:
+    """Static text card."""
+    return dcc.Markdown(text, dangerously_allow_html=False)
+
+
+@capture("card")
+def nav_card(data_frame: pd.DataFrame, text: str, href: str) -> dbc.Card:
+    """Static navigation card."""
+    return dbc.NavLink(
+        dcc.Markdown(text, dangerously_allow_html=False),
+        href=get_relative_path(href) if href.startswith("/") else href,
+    )
+
+
+# Example 1: KPI Card with Markdown
+# (+) Allows for unlimited customisation on text
+# (-) Custom styling becomes difficult due to className provision
+@capture("card")
+def kpi_card_mkd(data_frame: pd.DataFrame, title: str, value: str, agg_fct: Callable = sum) -> dbc.Card:
+    """Dynamic text card in form of a KPI Card."""
+    # LQ: Think about exposing an argument that allows for custom formatting such as formatting as currency
+    value = round(agg_fct(data_frame[value]), 2)
+
+    return dcc.Markdown(
+        f"""
+        ## {title}
+
+        # {value}
+        """,
+        dangerously_allow_html=False,
+    )
+
+
+# Example 2: KPI Card with HTML
+# (-) Customisation on text requires a new captured Callable to be created
+# (+) Allows for indefinite custom styling
+
+
+# LQ: Do we want all of these arguments? `title` and `value` are required, but the rest is extra functionality that
+# we could also outsource to creating their own CapturedCallable with that logic. It would be just more cumbersome to create these.
+@capture("card")
+def kpi_card(
+    data_frame: pd.DataFrame,
+    title: str,
+    value: str,
+    icon: Optional[str] = None,
+    agg_fct: Callable = sum,
+    value_format: Optional[str] = None,
+) -> dbc.Card:
+    """Dynamic text card in form of a KPI Card."""
+    value = agg_fct(data_frame[value])
+
+    return [
+        html.Div(
+            [
+                html.P(icon, className="material-symbols-outlined") if icon else None,
+                html.H2(title),
+            ],
+        ),
+        html.P(value_format.format(value) if value_format else value),
+    ]
+
+
+# LQ: Not sure if the removal of classNames is a better approach. It seems more unstable as it depends
+# on the component hierarchy not changing now. I slightly prefer to explictly provide classNames to the subcomponents here.
+
+
+# LQ: # In case we do want to go with the `value_format` argument - what's the best way to handle this for e.g. for the kpi_card_ref.
```

Comment:
My rough version of this below:
```
@capture("card")
def kpi_card_compare(
    data_frame: pd.DataFrame,
    column: str,  # Renamed value so it's like in filter,
    title: Optional[str] = None,  # Optional - use column by default like in filter
    icon: Optional[str] = None,
    agg_func: str = "sum",  # func, not fct. This takes strings which works better for YAML specification - see
    # https://stackoverflow.com/questions/65877567/passing-function-names-as-strings-to-pandas-groupby-aggregrate.
    value_format: str = "{value}",  # Default value means "no special formatting"
    # All above are same arguments, these two are new:
    reference_column: str,
    comparison_format: str = "{delta_relative:.1%} vs. reference ({reference_value})", # Note you do percentage
        # calculation using Python formation language itself: https://docs.python.org/3/library/string.html#format-specification-mini-language

) -> dbc.Card:
    """Dynamic text card in form of a KPI Card."""
    # Next line is pseudocode but it's easy to do in pandas
    value, reference_value = data_frame[column, reference_column].agg(agg_func)
    title = title or column.title()

    # Check logic here - it seemed wrong way round to me before
    delta = value - reference_value
    delta_relative = delta / reference_value # No % calculation here, it's done in comparison_format

    # Variables available in both comparison_format and value_format are value, reference_value, delta, delta_relative
    return [
        html.Div(
            [
                html.P(icon, className="material-symbols-outlined") if icon else None,
                html.H2(title),
            ],
        ),
        html.P(value_format.format(value=value, reference_value=reference_value, delta=delta,
                                   delta_relative=delta_relative)),
        html.Span(
            [
                html.Span(
                    "arrow_circle_up" if delta > 0 else "arrow_circle_down", className="material-symbols-outlined"
                ),
                html.Span(comparison_format.format(value=value, reference_value=reference_value, delta=delta,
                                                   delta_relative=delta_relative),
            ],
            className="delta-pos" if delta > 0 else "delta-neg",
        ),
    ]

    # TBD: names like
    # value, value_format, column
    # reference, reference_format or comparison_format, reference_column

    # TBD merge these two functions into just one? Since the second is just the same as the first with two additional
    # arguments. Not sure if good idea or not.
    # Can we remove any arguments easily? DOn't want them to get too complicated.
```

Reply:
We maybe also need an argument `positive_delta_is_good: bool = True`(not sure about name) as per https://github.com/mckinsey/vizro/pull/505#discussion_r1624754724.

Reply:
In short, I don't like the plotly API here and think it doesn't fit well into how we do other things. If we think we can cover most of the common functionality in just a couple of arguments `value_format` and `comparison_format` then I'd much rather do that instead of having `dict` arguments that nest various properties.

Possibly we can drop some of our arguments from an initial version if we don't think they'll be commonly used, but it's ok by me if we don't.